### PR TITLE
Trapezoidal (n=2)

### PR DESCRIPTION
**Pull request by MathisGD**

### CODE REVIEW

**Review by Jean-Grimal — APPROVED** [review]
very clean

**Review by MerlinEgalite — APPROVED** [review]

File: src/SpeedJumpIrm.sol [hunk 1]
```diff
@@ -147,24 +143,24 @@ contract AdaptiveCurveIrm is IIrm {
...
-                int256 step = linearAdaptation / N_STEPS;
-                // Compute the terms 1 to N_STEPS - 1.
-                for (int256 k = 1; k < N_STEPS; k++) {
-                    sumRateAtTarget += _newRateAtTarget(startRateAtTarget, step * k);
-                }
+                // The integral is approximated with the trapezoidal rule:
+                // avg ~= 1/T Σ_i=1^N (f((i-1) * T/N) + f(i * T/N)) / 2 * T/N
+                // Where f(x) = curve(startRateAtTarget * exp(speed * x), err).
+                // avg ~= 1/N * ( (f(0)+f(T))/2 + Σ_i=1^(N-1) f(i*T/N) )
+                // avg ~= 1/N * ( (f(startRateAtTarget)+f(endRateAtTarget))/2 + Σ_i=1^(N-1) curve(startRateAtTarget *
```
Comment (QGarchery) [inline src/SpeedJumpIrm.sol:150]:
```suggestion
                // avg ~= 1/N * ( (f(0)+f(T))/2 + Σ_i=1^(N-1) curve(startRateAtTarget *
```

**Review by QGarchery — COMMENTED** [review]
I think I have a way to make it a bit more concise, opening a PR on top

**Review by peyha — APPROVED** [review]
The calcul is correct for me

**Review by Rubilmax — APPROVED** [review]
I'm not sure all these detailed comments are necessary to be frank, but they are obviously welcomed to make the review the easiest

**Review by MerlinEgalite — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Review by QGarchery — APPROVED** [review]

**Pull request merged**